### PR TITLE
Force Algorithmically dark mode

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/715106103902962/1206630026838095/f

### Description
This PR adds support for [algorithmically forced dark mode.](https://developer.android.com/reference/androidx/webkit/WebSettingsCompat#setAlgorithmicDarkeningAllowed(android.webkit.WebSettings,boolean))

### Steps to test this PR

_Dark mode_
- [x] Launch app and enable dark theme.
- [x] Open http://permission.site.
- [x] Verify site is displayed in light theme.
- [x] Open Settings / Appearance and enable Force Dark Mode.
- [x] Restart the app so changes take effect.
- [x] Open http://permission.site.
- [x] Verify site is displayed in dark theme.

### UI changes
| Before  | After |
| ------ | ------
|Screenshot_20240605_125706|Screenshot_20240605_125744|

### CODE REVIEW

**Comment by malmstein** [comment]
* **#4614** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4614?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @malmstein and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/main/java/com/duckduckgo/app/appearance/AppearanceViewModel.kt [hunk 1]
```diff
@@ -63,6 +65,8 @@ class AppearanceViewModel @Inject constructor(
                 currentViewState().copy(
                     theme = themingDataStore.theme,
                     appIcon = settingsDataStore.appIcon,
+                    forceDarkModeEnabled = settingsDataStore.experimentalWebsiteDarkMode,
+                    canForceDarkMode = themingDataStore.theme != DuckDuckGoTheme.LIGHT,
```
Comment (CrisBarreiro) [inline app/src/main/java/com/duckduckgo/app/appearance/AppearanceViewModel.kt:69]: I think we should also check is theme is SYSTEM_DEFAULT, and system default turns out to be light

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt [hunk 2]
```diff
@@ -2574,6 +2585,12 @@ class BrowserTabFragment :
         settings.databaseEnabled = false
     }
 
+    @Suppress("NewApi") // This API and the behaviour described only apply to apps with targetSdkVersion ≥ TIRAMISU.
```
Comment (CrisBarreiro) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:2588]: Do we need this? Our target is Upside Down Cake already

File: app/src/main/java/com/duckduckgo/app/appearance/AppearanceViewModel.kt [hunk 3]
```diff
@@ -106,4 +110,8 @@ class AppearanceViewModel @Inject constructor(
     private fun currentViewState(): ViewState {
         return viewState.value
     }
+
+    fun onForceDarkModeSettingChanged(checked: Boolean) {
+        settingsDataStore.experimentalWebsiteDarkMode = checked
```
Comment (CrisBarreiro) [inline app/src/main/java/com/duckduckgo/app/appearance/AppearanceViewModel.kt:115]: We shouldn't be calling this from the main thread

**Review by CrisBarreiro — CHANGES_REQUESTED** [review]
Looks good overall, feel free to ignore minor issues, but I think we should address the edge case on system_default

**Review by malmstein — COMMENTED** [review]
Thanks @CrisBarreiro! I’ve added the edge case verification for system theming as suggested.

**Review by CrisBarreiro — APPROVED** [review]

**Pull request merged**